lxd/storage/drivers/driver_lvm_volumes.go

```diff
@@ -548,10 +564,17 @@ func (d *lvm) SetVolumeQuota(vol Volume, size string, allowUnsafeResize bool, op
 			return err
 		}
 
-		// Move the VM GPT alt header to end of disk if needed (not needed in unsafe resize mode as it is
-		// expected the caller will do all necessary post resize actions themselves).
-		if vol.IsVMBlock() && !allowUnsafeResize {
-			// Activate the volume for resizing.
+		// The new blocks in a grown volume will need clearing if using a thick pool.
+		needsClearing := !d.usesThinpool() && (oldSizeBytes < sizeBytes)
```

Comment:
So this deals with zeroing the added bytes to a fat LV being grown. What about the shrinking case? I don't know if LVM thin does discard the freed block when shrinking but it's something we should be doing to avoid leaving customer data on disk.

Reply:
This issue is about not exposing old data to newly created volumes, rather than securely deleting data off of the disk when the volume is deleted/shrunk, so I think this doesn't need to be addressed in this PR.